cpp/src/util/main/UnitTests.cpp

```diff
@@ -125,6 +136,78 @@ TEST(eigen_util, sort_columns) {
   }
 }
 
+TEST(eigen_util, sort_columns_one_element) {
+  constexpr int kNumRows = 1;
+  constexpr int kMaxNumCols = 1;
+  using Array = Eigen::Array<float, kNumRows, Eigen::Dynamic, 0, kNumRows, kMaxNumCols>;
+
+  Array array{
+      {3},
+  };
+
+  array = eigen_util::sort_columns(array);
+
+  Array expected{
+      {3},
+  };
+
+  for (int r = 0; r < array.rows(); r++) {
+    for (int c = 0; c < array.cols(); c++) {
+      EXPECT_EQ(array(r, c), expected(r, c));
+    }
+  }
+}
+
+TEST(eigen_util, extract_dim_v) {
+  using Sizes = Eigen::Sizes<123, 456, 789>;
+  constexpr int N = eigen_util::extract_dim_v<1, Sizes>;
+  EXPECT_EQ(N, 456);
+
+}
```

Comment:
Minor nitpick: please remove the blank newline here.

Note that you could simply check this at compile time by using `static_assert()`, since the quantities you are comparing are both compile-time constants. It's still fine to keep this test though.